services/searcher.ts

```diff
@@ -458,6 +459,60 @@ class Searcher {
     };
   }
 
+  filterSection(
+    curSection: Section,
+    honorsFilter: boolean,
+    campusFilter: string
+  ): boolean {
+    if (honorsFilter && !curSection.honors) {
+      return false;
+    }
+    if (campusFilter !== "" && curSection.campus !== campusFilter) {
+      return false;
+    }
+
+    return true;
+  }
+
+  filterResults(filters, results): SearchResult[] {
+    // Do extra filtering if Honors or Campus Filters are being applied since those filter sections,
+    // which ElasticSearch doesn't actually filter out.
+
+    const validFilters = this.validateFilters(filters);
+
+    const honorsFilter: boolean = validFilters["honors"] ? true : false;
+
+    const campusFilter: string = "" + (validFilters["campus"] ?? "");
+
+    // create a new list initially containing the results, if we are doing further filtering
+    // we'll set it to an empty list and build up the new filtered results.
+    let filteredResults: SearchResult[] = results;
+    if (honorsFilter || campusFilter !== "") {
+      filteredResults = [];
```

Comment:
```suggestion
    if (!honorsFilter && campusFilter === "") {
      return results;
    }
    .... // handle the filtering here
```

Instead of wrapping the "we need to filter further" case in an `if` block, just write a simple `if` to see if we DON'T need to filter further. This is preferred because:
- we don't have to wrap a large chunk of code in a block - it's simpler & easier to read (and less indentation)
- instead of assigning `filteredResults` to `results` and conditionally changing that, just assign `const filteredResults = []`. It's simpler, doesn't require a comment explaining it, and allows you to use `const`